src/meson.build

```diff
@@ -0,0 +1,61 @@
+###########
+# Library #
+###########
+
+spoa_sources = files([
+  'alignment_engine.cpp',
+  'graph.cpp',
+  'simd_alignment_engine_dispatcher.cpp',
+  'sisd_alignment_engine.cpp',
+])
+
+spoa_simd_lib = []
+
+if (get_option('spoa_generate_dispatch'))
+  simd = import('unstable_simd')
+
+  spoa_simd_sources = files([
+    'simd_alignment_engine_dispatch.cpp',
+  ])
+
+  spoa_simd_lib += simd.check(
+    'spoa',
+    sse2 : spoa_simd_sources,
+    sse41 : spoa_simd_sources,
+    avx2 : spoa_simd_sources,
+    dependencies : spoa_lib_deps,
+    include_directories : spoa_include_directories,
+    cpp_args : [spoa_warning_flags, spoa_lib_flags],
+    compiler : cpp
+  )[0]
+endif
+
+spoa_lib_install = (not meson.is_subproject()) or (get_option('default_library') != 'static')
+
+spoa_lib = library(
+  'spoa',
+  spoa_sources,
+  soversion : meson.project_version(),
+  version : meson.project_version(),
+  install : spoa_lib_install,
+  dependencies : spoa_lib_deps,
+  include_directories : spoa_include_directories,
+  link_with : spoa_simd_lib,
+  cpp_args : [spoa_warning_flags, spoa_lib_flags])
+
+##############
+# Executable #
+##############
+
+if (get_option('spoa_build_exe'))
+  spoa_exe_flags = ['-DVERSION="' + meson.project_version() + '"']
```

Comment:
promise to change it to a `spoa_config.h.in` in both meson and cmake in a followup PR?